pkg/utils/pod.go

```diff
@@ -112,15 +112,12 @@ func IsCriticalPod(pod *v1.Pod) bool {
 	if IsMirrorPod(pod) {
 		return true
 	}
-	if pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority) {
```

Comment:
This needs to stay here. You should never be allowed to evict critical pods.

Reply:
Isn't this check basically replaced by IsPodEvictableBasedOnPriority?

Reply:
Given the way it's implemented:

```
func IsPodEvictableBasedOnPriority(pod *v1.Pod, priority int32) bool {
	return pod.Spec.Priority == nil || *pod.Spec.Priority < priority
}
```

a user can choose any int32 value. Even the one greater than the system critical priority. Thus, if improperly configured the descheduler can evict even the critical pods. `return pod.Spec.Priority == nil || (*pod.Spec.Priority < priority && *pod.Spec.Priority < SystemCriticalPriority)` would do a bit better.

My point is `IsEvictable` function is supposed to be static. The hardline for deciding when a pod is evictable or not without any degree of flexibility. If `IsEvictable` decides a pod is not evictable, it is not. It's a safe guard condition that can not be changed from the outside (except for setting some true/false flags).

Reply:
Good points, I did notice this change at first too but thought it was just being covered by the new code.

In that case, we should probably keep it the same as it was right?

Reply:
@lixiang233 what do you think?

Reply:
Agree, if someone wants to evict system-critical pods, he can add `descheduler.alpha.kubernetes.io/evict` annotation. I'll add `*pod.Spec.Priority < SystemCriticalPriority` to `IsPodEvictableBasedOnPriority`.